src/main/kotlin/no/skatteetaten/aurora/gobo/integration/boober/AuroraConfigService.kt

```diff
@@ -100,15 +101,28 @@ class AuroraConfigService(
         token: String,
         details: ApplicationDeploymentDetailsResource,
         applyLink: String
-    ): JsonNode {
+    ): RedeployResponse {
         val payload = ApplyPayload(listOf(details.applicationDeploymentCommand.applicationDeploymentRef))
-        return booberWebClient.put<JsonNode>(token, applyLink, body = payload).toMono().blockNonNullWithTimeout()
+        val json = booberWebClient.put<JsonNode>(token, applyLink, body = payload).toMono().blockNonNullWithTimeout()
+        return RedeployResponse(json)
     }
 
     private fun <T> Mono<T>.blockNonNullWithTimeout() =
         this.blockNonNullAndHandleError(Duration.ofSeconds(30), "boober")
 }
 
+data class RedeployResponse(private val json: JsonNode) {
+
+    val applicationDeploymentId: String
+        @JsonProperty("id")
+        get() = json.at("/applicationDeploymentId").textValue()
+            ?: throw IllegalStateException("No applicationDeploymentId found in response")
```

Comment:
Kan dette skje? Og evt når vil det skje? Det vil egentlig kanskje si at mokey feilet?

Reply:
Denne responsen kommer fra boober apply/deploy. Det er nok svært usannsynlig at den responsen ikke inneholder affiliation, og hvis den gjør det så er det nok noe veldig feil hos boober. Men jeg tenkte uansett at vi bør håndtere det og bare kaste en feil i gobo, i stedet for at vi får en kryptisk feilmelding ut.

Reply:
da er jeg enig.